src/features/delegation/sagas/requestPrivateKey.saga.ts

```diff
@@ -5,13 +5,44 @@ import requestDelegatedPrivateKey from '$/utils/requestDelegatedPrivateKey'
 import { error, success } from '$/utils/toaster'
 import takeEveryUnique from '$/utils/takeEveryUnique'
 import { WalletAction } from '$/features/wallet'
+import isAuthorizedDelegatedAccount from '$/utils/isAuthorizedDelegatedAccount'
+import { Wallet } from 'ethers'
+import authorizeDelegatedAccount from '$/utils/authorizeDelegatedAccount'
+import { toast } from 'react-toastify'
 
 function* onRequestPrivateKeyAction({
     payload: { owner, provider },
 }: ReturnType<typeof DelegationAction.requestPrivateKey>) {
     try {
         const privateKey: string = yield requestDelegatedPrivateKey(provider, owner)
 
+        const isDelegationAuthorized: boolean = yield isAuthorizedDelegatedAccount(
+            owner,
+            new Wallet(privateKey).address,
+            provider
+        )
+
+        if (!isDelegationAuthorized) {
+            let toastId
+
+            try {
+                toastId = toast.loading('Authorizing your delegated wallet...', {
+                    position: 'bottom-left',
+                    autoClose: false,
+                    type: 'info',
+                    closeOnClick: false,
+                    hideProgressBar: true,
+                })
+                yield authorizeDelegatedAccount(owner, privateKey, provider)
```

Comment:
If you reject this transaction the code will continue on to "Access delegated successfully." if the priv key is there.